### PR TITLE
chore (proof-mpc): Move `mpc-protocols` crate into `pera-core`

### PR DESCRIPTION
**Pull request by ItayLevyOfficial**
We needed to do it to be able to inject the validators signature MPC logic into Pera while avoiding import cycles.

### CODE REVIEW

File: crates/pera-analytics-indexer/Cargo.toml [hunk 1]
```diff
@@ -51,7 +51,7 @@ pera-json-rpc-types.workspace = true
 pera-package-resolver.workspace = true
 simulacrum.workspace = true
 arrow.workspace = true
-gcp-bigquery-client = "0.18.0"
+gcp-bigquery-client = "0.23.0"
```
Comment (ItayLevyOfficial) [inline crates/pera-analytics-indexer/Cargo.toml:54]: Needed to change this as this package fails to compile otherwise

**Review by erik-3milabs — APPROVED** [review]
LGTM!

**Review by cdesaintguilhem — APPROVED** [review]
Looks clean to me.

**Comment by ycscaly** [comment]
Why is this PR needed? why not change `244` so that the packages would be created correctly instead of creating something just to change it?

**Comment by ItayLevyOfficial** [comment]
> Why is this PR needed? why not change `244` so that the packages would be created correctly instead of creating something just to change it?

Because two people already reviewed it and changing it would discard their comments. Refactor PRs, like this one, are common, and this PR is quite clean

**Pull request closed without merging**